src/display/node_stream.js

```diff
@@ -0,0 +1,374 @@
+/* Copyright 2012 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* globals __non_webpack_require__ */
+
+let fs = __non_webpack_require__('fs');
+let http = __non_webpack_require__('http');
+let https = __non_webpack_require__('https');
+let url = __non_webpack_require__('url');
+
+import { assert, createPromiseCapability } from '../shared/util';
+import { validateRangeRequestCapabilities } from './network_utils';
+
+class PDFNodeStream {
+  constructor(options) {
+    this.options = options;
+    this.source = options.source;
+    this.url = url.parse(this.source.url);
+    this.isHttp = this.url.protocol === 'http:' ||
+                  this.url.protocol === 'https:';
+    this.isFsUrl = !this.url.host;
```

Comment:
Is `isFsUrl` supposed to say whether a URL refers to the filesystem?
If so, this is not the right way to say so. The `file:` scheme is commonly used to refer to local files, and it can have a host part too - see e.g. https://en.wikipedia.org/wiki/File_URI_scheme

What are the expected values for url that you intend to support here?